test/policies/stock_policy_test.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class StockPolicyTest < ActiveSupport::TestCase
+  test "factory" do
+    assert build(:stock).validate!
+  end
+
+  test "index? allows any logged-in user and denies guests" do
+    stock = build(:stock)
+
+    assert_permit create(:admin), stock, :index
+    assert_permit create(:teacher), stock, :index
+    assert_permit create(:student), stock, :index
+
+    refute_permit nil, stock, :index
+  end
+
+  test "show? allows admins and teachers to view any stock" do
+    active_stock = create(:stock, archived: false)
+    archived_stock = create(:stock, archived: true)
+
+    admin = create(:admin)
+    teacher = create(:teacher)
+
+    # Admin
+    assert_permit admin, active_stock, :show
+    assert_permit admin, archived_stock, :show
+
+    # Teacher
+    assert_permit teacher, active_stock, :show
+    assert_permit teacher, archived_stock, :show
+  end
+
+  test "show? allows students to view only active stocks and denies guests" do
+    active_stock = create(:stock, archived: false)
+    archived_stock = create(:stock, archived: true)
+
+    student = create(:student)
+
+    assert_permit student, active_stock, :show
+    refute_permit student, archived_stock, :show
+
+    refute_permit nil, active_stock, :show
+    refute_permit nil, archived_stock, :show
+  end
+
+  test "CRUD actions are allowed only for admins" do
+    stock = build(:stock)
+
+    admin = create(:admin)
+    teacher = create(:teacher)
+    student = create(:student)
+
+    # Admin allowed
```

Comment:
```suggestion
```
We don't need these comments.  The tests are self documenting and it's obvious from the test that admin is allowed also you've documented by naming the test something descriptive.